zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -477,10 +478,20 @@ void closeInterpreters(String user, String noteId) {
 
   public void close() {
     LOGGER.info("Close InterpreterSetting: " + name);
-    for (ManagedInterpreterGroup intpGroup : interpreterGroups.values()) {
-      intpGroup.close();
-    }
+    List<Thread> closeThreads = interpreterGroups.values().stream()
+            .map(g -> new Thread(g::close, name + "-close"))
+            .peek(Thread::start)
```

Comment:
Предлагаю добавить потокам UncaughtExceptionHandler с логгером.
```suggestion
            .peek(t -> t.setUncaughtExceptionHandler((th, e) ->
                LOGGER.error("InterpreterSetting close error", e)))
            .peek(Thread::start)
```

Мотивация простая, если в одном из `close()` вылетал не обработанный `Exception`, то раньше он подымался до самого [main()](https://github.com/TinkoffCreditSystems/zeppelin/blob/becfdf15f6cc80669fe70ae4443a21c807946827/zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java#L289-L298) и там попадал в лог. А в новой версии можем его потерять...

Reply:
Поправил в 4bab30f76f302b30227d62377cd735bf1c723896.